tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/ParticipatingTckResource.java

```diff
@@ -0,0 +1,126 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.eclipse.microprofile.lra.tck.participant.api;
+
+import org.eclipse.microprofile.lra.annotation.Compensate;
+import org.eclipse.microprofile.lra.annotation.Complete;
+import org.eclipse.microprofile.lra.annotation.ws.rs.LRA;
+import org.eclipse.microprofile.lra.annotation.ws.rs.Leave;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Logger;
+
+import static org.eclipse.microprofile.lra.annotation.ws.rs.LRA.LRA_HTTP_HEADER;
+
+@ApplicationScoped
+@Path(ParticipatingTckResource.TCK_PARTICIPANT_RESOURCE_PATH)
+public class ParticipatingTckResource {
+    private static final Logger LOGGER = Logger.getLogger(ParticipatingTckResource.class.getName());
+
+    public static final String TCK_PARTICIPANT_RESOURCE_PATH = "participating-tck-resource";
+    public static final String JOIN_WITH_EXISTNG_LRA_PATH = "/join-with-existing-lra";
+    public static final String JOIN_WITH_NEW_LRA_PATH = "/join-with-new-LRA";
+    public static final String COMPLETED_CNT_PATH = "/completed-count";
+    public static final String COMPENSATED_CNT_PATH = "/compensated-count";
+    public static final String LEAVE_PATH = "/leave";
+
+    private static final AtomicInteger COMPLETED_COUNT = new AtomicInteger(0);
+    private static final AtomicInteger COMPENSATED_COUNT = new AtomicInteger(0);
+
+    @PUT
+    @Path("/compensate")
+    @Compensate
+    public Response compensateWork(@HeaderParam(LRA_HTTP_HEADER) String lraId, String userData)
+            throws NotFoundException {
+        if(lraId == null) {
+            throw new NullPointerException("lraId can't be null as it should be invoked with the context");
+        }
+
+        COMPENSATED_COUNT.incrementAndGet();
+
+        LOGGER.info(String.format("LRA id '%s' was compensated", lraId));
+
+        return Response.ok().build();
+    }
+
+    @PUT
+    @Path("/complete")
+    @Complete
+    public Response completeWork(@HeaderParam(LRA_HTTP_HEADER) String lraId, String userData)
+            throws NotFoundException {
+        if(lraId == null) {
+            throw new NullPointerException("lraId can't be null as it should be invoked with the context");
+        }
+
+        COMPLETED_COUNT.incrementAndGet();
+
+        LOGGER.info(String.format("LRA id '%s' was completed", lraId));
+
+        return Response.ok().build();
+    }
+
+    @PUT
+    @Path(LEAVE_PATH)
+    @LRA(value = LRA.Type.SUPPORTS, end = false)
```

Comment:
having `@LRA` on `@Leave` method may be confusing for readers of the spec. Is it necessary to have it here?

Reply:
-1 how else would the implementation know which LRA the participant is leaving.

Reply:
by LRA_HTTP_HEADER?

Reply:
You should read the specification, in particular the javadoc for LRA may help you understand.

Reply:
OK, I did read it again but I don't understand what you mean or pointing to still. Yes it's written there that you can use HeaderParam to inject context but that prohibits injection in anything which is not annotated by `@LRA`?

By the way, good that you pointed me to the javadoc -- [1, 2] should be also adjusted for this PR

[1] https://github.com/eclipse/microprofile-lra/blob/master/api/src/main/java/org/eclipse/microprofile/lra/annotation/ws/rs/LRA.java#L57
[2] https://github.com/eclipse/microprofile-lra/blob/master/api/src/main/java/org/eclipse/microprofile/lra/annotation/ws/rs/LRA.java#L74

The javadoc for `@Leave` is also not mentioning how implementation should propagate information about which LRA is concerned to the users it only mentions that current active context of the LRA should the participant left. I think we should allow this trough http header same way as for complete/compensate methods.

Reply:
Raise your concerns elsewhere, this PR is to remove the LRAClient dependency.

Reply:
@mmusgrov no need to be rude. My original concern is the removal of line 89 if possible and if not to create a separate method for `@LRA(value = LRA.Type.SUPPORTS, end = false)` to not mix participant marker annotation with `@LRA`

Reply:
Oh you are talking about the presence of LRAClient in the header, that's fair enough. But can you do a single review with your concerns rather than coming back multiple times with new issues.

Reply:
Actually you are reviewing the code on the master branch, my PR has made changes to master. So can I ask you to review the changes
- made in this PR (not what's on master)
- try to do it in one pass

Reply:
Asking you to only review what's in the PR is not being rude.

Reply:
I always do a bulk of suggestions as I did yesterday 2:21 PM in this PR ;) This conversation is only reactions to your comments. The javadoc in `@LRA` comments came also as such reaction but you are right I will look them up in the PR.

Reply:
No I mean you are reviewing code which isn't even in this PR, you said:

> By the way, good that you pointed me to the javadoc -- [1, 2] should be also adjusted for this PR

but those two lines are not in this PR they are only on the master branch. So if you limit yourself to reviewing just what's in this PR then we can make swifter progress.